xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/UserMentionNotificationService.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.mentions.MentionNotificationService;
+import org.xwiki.mentions.MentionsConfiguration;
+import org.xwiki.mentions.events.MentionEvent;
+import org.xwiki.mentions.events.MentionEventParams;
+import org.xwiki.mentions.notifications.MentionNotificationParameter;
+import org.xwiki.mentions.notifications.MentionNotificationParameters;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.observation.ObservationManager;
+
+import static java.util.Collections.singleton;
+import static org.xwiki.mentions.internal.UserMentionNotificationService.TYPE;
+
+/**
+ * Default implementation of {@link MentionNotificationService}.
+ *
+ * @version $Id$
+ * @since 12.5RC1
+ */
+@Component
+@Named(TYPE)
+@Singleton
+public class UserMentionNotificationService implements MentionNotificationService
+{
+    /**
+     * Type of the component.
+     */
+    public static final String TYPE = "user";
```

Comment:
Well, here it's the default implem for XWiki Users, right? "user" sounds a bit too generic to me.